### PR TITLE
Users can view their Grade object for each course

### PR DESCRIPTION
**Pull request by ecoist-ste**
1. **CourseGradeManager:** performs the Canvas API calls to fetch a user's active enrollments stored in an array of type Enrollment. An enrollment object contains a Grade object where we can obtain scores and grade information.
2. **CourseGradeView:** presents the properties (_currentScore, currentGrade, finalScore, finalGrade_) of a Grade object for each course
3. Note that some courses don't have a fully available Grade object. For example, a course instructor hasn't started calculating the current score or hasn't given any graded assignment. 

https://github.com/user-attachments/assets/531748cb-7542-4e5a-9909-a29f4cdb704a

### CODE REVIEW

**Review by rahulon12 — COMMENTED** [review]
Do you want to rebase from main and integrate Aziz's changes from #6?

**Comment by azooz2003-bit** [comment]
@ecoist-ste You could do this as a separate PR, but let's have a look at gradebooks? (assignments, quizzes, etc.)

File: CanvasPlusPlayground/Views/CourseGradeView.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            
+            if let finalGrade = enrollment?.grades?.finalGrade {
+                Text("Current Score: \(finalGrade)")
+            } else {
+                Text("Final Grade unavailable")
+            }
+           
+        }
+        .task {
+            await gradeManager.fetchEnrollments()
```
Comment (azooz2003-bit) [inline CanvasPlusPlayground/Views/CourseGradeView.swift:49]: @ecoist-ste Is it possible that we fetch enrollments at the `CourseListView`? Then we'd have less repetitive work - I'm assuming the enrollments are the same as courses.

**Review by azooz2003-bit — CHANGES_REQUESTED** [review]
Small comment! All else LGTM! 😉

**Comment by ecoist-ste** [comment]
> Do you want to rebase from main and integrate Aziz's changes from #6?

Yes I will do that

File: CanvasPlusPlayground/Views/CourseGradeView.swift (hunk 1 above)
Comment (ecoist-ste) [reply CanvasPlusPlayground/Views/CourseGradeView.swift:49]: That sounds good. I will do that instead

**Pull request closed without merging**